integration/dashboards_test.go

```diff
@@ -111,6 +112,48 @@ func TestDashboardCreateWithCustomGraph(t *testing.T) {
 	assertDashboardEquals(t, actual, expected)
 }
 
+func TestDashboardGetWithNewId(t *testing.T) {
+	expected := getTestDashboard(createGraph)
+	// create the dashboard and compare it
+	actual, err := client.CreateDashboard(expected)
+	if err != nil {
+		t.Fatalf("Creating a dashboard failed when it shouldn't. (%s)", err)
+	}
+
+	defer cleanUpDashboard(t, *actual.Id)
+
+	assertDashboardEquals(t, actual, expected)
+
+	// try to fetch it freshly using the new id format and compare it again
+	actual, err = client.GetDashboard(*actual.NewId)
+	if err != nil {
+		t.Fatalf("Retrieving a dashboard failed when it shouldn't. (%s)", err)
+	}
+	assertDashboardEquals(t, actual, expected)
```

Comment:
We could also test that `Id` and `NewId` are the same (between the 2 objects returned by `CreateDashboard`).